runsc/cmd/update.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2018 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/google/subcommands"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"gvisor.dev/gvisor/runsc/cmd/util"
+	"gvisor.dev/gvisor/runsc/config"
+	"gvisor.dev/gvisor/runsc/container"
+	"gvisor.dev/gvisor/runsc/flag"
+)
+
+func i64Ptr(i int64) *int64   { return &i }
+func u64Ptr(i uint64) *uint64 { return &i }
+func u16Ptr(i uint16) *uint16 { return &i }
+func boolPtr(b bool) *bool    { return &b }
+
+// Update implements subcommands.Command for the "update" command.
+type Update struct {
+	resources string
+
+	cpuPeriod    uint64
+	cpuQuota     int64
+	cpuBurst     uint64
+	cpuShares    uint64
+	cpuRtPeriod  uint64
+	cpuRtRuntime int64
+	cpusetCpus   string // can this be string or this has to be list of smth?
+	cpusetMems   string
+	cpuIdle      int64
+
+	memory            int64
+	memoryReservation int64
+	memorySwap        int64
+
+	blkioWeight int
+
+	pidsLimit int64
+
+	l3CacheSchema string
+	memBwSchema   string
+}
+
+// Name implements subcommands.Command.Name.
+func (*Update) Name() string {
+	return "update"
+}
+
+// Synopsis implements subcommands.Command.Synopsis.
+func (*Update) Synopsis() string {
+	return "update container resource constraints"
+}
+
+// Usage implements subcommands.Command.Usage.
+func (*Update) Usage() string {
+	return `update [flags] <container id> - update container resource constraints
+`
+}
+
+// SetFlags implements subcommands.Command.SetFlags.
+func (u *Update) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&u.resources, "resources", "", `path to the file containing the resources to update or '-' to read from the standard input
```

Comment:
Note that the default value means that resources will be read from the remaining flags.